### ISSUE TITLE
query not returning all of the results (only in javascript)

### ISSUE THREAD

**Issue by katinas15** [influxdata/influxdb-client-js#236]
__Expected Behavior:__

Query should return correct number of results. Just like this query:

```javascript
const fluxQuery1 =
'from(bucket:"' + bucket + '")'
+ '|> range(start: -' + 30 + 'd) ' 
+ '|> group(columns: ["_measurement"])'
+ '|> count(column: "_value")' 

console.log('*** QUERY ROWS ***')
queryApi.queryRows(fluxQuery1, {
  next(row, tableMeta) {
    const o = tableMeta.toObject(row)
    console.log(o)
  },
  error(error) {
    console.error(error)
  },
  complete() {},
})
```
Which returns the correct number of results:
```
{
  result: '_result',
  table: 0,
  _start: '2020-07-16T13:29:39.521937171Z',
  _stop: '2020-08-15T13:29:39.521937171Z',
  _value: 434,
  _measurement: 'event'
}
```

__Current Behavior:__

Query returns less number of results than expected. Used same query just without  'count()' and I do the counting in javascript:
```javascript
const fluxQuery2 =
'from(bucket:"' + bucket + '")'
+ '|> range(start: -' + 30 + 'd) ' 
+ '|> group(columns: ["_measurement"])'
+ '|> sort(columns: ["_time"], desc: true)'

queryApi.collectRows(fluxQuery2)
.then(data => {
    console.log(data.length)
})
.catch(error => {
    console.error(error)
})
```
Which returns the incorrect number of results (result changes every time I run the code):
```
231
188
190
...
```
Same results occur when using 'queryRows' instead of 'collectRows'.
I have tried the same thing in Python and it produces the correct result. It is most likely an issue with javascript, query completing before it is loaded into a variable ?

__Specifications:__

- Version: 1.6.0
- Platform: windows 10

**Comment by sranka** [comment]
Thank you for reporting an issue. Unfortunately, I am not able to reproduce the same locally. Can you please attach what is on the wire during the communication with InfluxDB? You can also use the following bash script:

```
#!/bin/bash
INFLUXDB_ORG=my-org
INFLUXDB_URL=http://localhost:9999
INFLUXDB_TOKEN=my-token
INFLUXDB_BUCKET=my-bucket

function query {
  POST_BODY='{"type":"flux","dialect":{"header":true,"delimiter":",","commentPrefix":"#","annotations": ["datatype", "group", "default"]}, "query": "'${1//\"/\\\"}'"}'
  curl -H "Content-Type: application/json" -H "Accept: application/json" -H "Authorization: Token ${INFLUXDB_TOKEN}" -X POST -d "$POST_BODY" $INFLUXDB_URL/api/v2/query?org=$INFLUXDB_ORG
}

query 'from(bucket:"'${INFLUXDB_BUCKET}'") |> range(start: -30d) |> group(columns: ["_measurement"]) |> count(column: "_value")'
query 'from(bucket:"'${INFLUXDB_BUCKET}'") |> range(start: -30d) |> group(columns: ["_measurement"]) |> sort(columns: ["_time"], desc: true)'
```

**Comment by katinas15** [comment]
Hey, thanks for the reply. You mean I should attach the start of the output?:
```
#datatype,string,long,dateTime:RFC3339,dateTime:RFC3339,long,string
#group,false,false,true,true,false,true
#default,_result,,,,,
,result,table,_start,_stop,_value,_measurement
,,0,2020-07-17T08:15:10.098530157Z,2020-08-16T08:15:10.098530157Z,439,event

#datatype,string,long,dateTime:RFC3339,dateTime:RFC3339,dateTime:RFC3339,string,string,string,string,string,string
#group,false,false,true,true,false,false,false,true,false,false,false
#default,_result,,,,,,,,,,
,result,table,_start,_stop,_time,_value,_field,_measurement,postDate,tagid,type
```
Anyways, I also attached the full output of the script (this one also seems to work correctly).

[output.txt](https://github.com/influxdata/influxdb-client-js/files/5080272/output.txt)

**Comment by katinas15** [comment]
Right now I replaced my code with a POST request, instead of using 'collectRows' and it works perfectly fine! Thanks for the idea.

```javascript
      axios.post(url + '/api/v2/query?org=' + org,
          {
            query: fluxQuery2
          }, 
          {
              headers: {
                'Content-Type': 'application/json',
                'Accept': 'application/json',
                'Authorization': 'Token ' + token + ''
              }
          })
        .then(res => {
          res = res.data.toString()
          res = res.split('\n')
          console.log(res.length)
        })
        .catch(e => {
          console.log(e);
        });
```

**Comment by sranka** [comment]
NP, if such a workaround works for you. Can you do please do me a favor and send the full zipped output to pavel.zavora@bonitoo.io ... I would rather make myself sure whether/how this is affecting the users of this library.

**Comment by sranka** [comment]
Thank you @katinas15, I can reproduce it with the data that you provided, it has probably something to do with an encoded JSON. Can you use your data in an automated test?

**Comment by sranka** [comment]
@katinas15 Sorry for having trouble with this. It is fixed, it will be available for you in the next nightly build (on Moday). It can happen randomly under certain circumstances depending on the size of HTTP response chunks delivered by node's http library. Because the chunk size varies, it also explains why you did receive different results every next run.

**Comment by katinas15** [comment]
Thanks a lot for the fast response :)

**Comment by sranka** [comment]
The fix is available in 1.7.0-nightly.2
